podcast-player.js

```diff
@@ -0,0 +1,342 @@
+    this.duration = this.audio.duration;
+  }
+
+  handleTimeUpdate(e) {
+    this.currentTime = this.audio.currentTime;
+  }
+
+  timeJump(event) {
+    let params = new URLSearchParams(window.location.hash.substring(1));
+    let t = params.get("t") || 0;
+
+    console.log(window, params, t, event);
+
+    var timestamp = this.parseTime(t);
+
+    if (t) {
+      // Preload the media
+      this.audio.setAttribute("preload", "true");
+      // Set the current time. Will update if playing. Will fail if paused.
+      this.audio.currentTime = timestamp;
+      // If the media is able to play, play.
+      this.audio.addEventListener(
+        "canplay",
+        () => {
+          /* only start the player if it is not already playing */
+          if (!this.audio.paused) {
+            return false;
+          }
+
+          this.audio.currentTime = timestamp;
+          this.audio.play();
+          this.playing = true;
+        },
+        false
+      );
+    }
+  }
+
+  parseTime(str) {
+    var plain = /^\d+(\.\d+)?$/g,
+      npt = /^(?:npt:)?(?:(?:(\d+):)?(\d\d?):)?(\d\d?)(\.\d+)?$/,
+      quirks = /^(?:(\d\d?)[hH])?(?:(\d\d?)[mM])?(\d\d?)[sS]$/,
+      match;
+    if (plain.test(str)) {
+      return parseFloat(str);
+    }
+    match = npt.exec(str) || quirks.exec(str);
+    if (match) {
+      return (
+        3600 * (parseInt(match[1], 10) || 0) +
+        60 * (parseInt(match[2], 10) || 0) +
+        parseInt(match[3], 10) +
+        (parseFloat(match[4]) || 0)
+      );
+    }
+    return 0;
+  }
+
+  toHHMMSS(totalsecs) {
+    var sec_num = parseInt(totalsecs, 10); // don't forget the second param
+    var hours = Math.floor(sec_num / 3600);
+    var minutes = Math.floor((sec_num - hours * 3600) / 60);
+    var seconds = sec_num - hours * 3600 - minutes * 60;
+
+    if (hours < 10) {
+      hours = "0" + hours;
+    }
+    if (minutes < 10) {
+      minutes = "0" + minutes;
+    }
+    if (seconds < 10) {
+      seconds = "0" + seconds;
+    }
+
+    hours = hours > 0 ? hours + ":" : "";
+    minutes = minutes + ":";
+
+    var time = hours + minutes + seconds;
+    return time;
+  }
+
+  changeSpeed() {
+    this.currentSpeedIdx =
+      this.currentSpeedIdx + 1 < this.speeds.length
+        ? this.currentSpeedIdx + 1
+        : 0;
+    this.audio.playbackRate = this.speeds[this.currentSpeedIdx];
+  }
+
+  mute() {
+    this.audio.muted = !this.audio.muted;
+    this.muted = this.audio.muted;
+  }
+
+  play() {
+    if (this.audio.paused) {
+      console.log(this.currentTime, this.audio.currentTime);
+      this.audio.play();
+    } else {
+      this.audio.pause();
+    }
+    this.playing = !this.audio.paused;
+  }
+
+  stop() {
+    this.playing = !this.audio.paused;
+  }
+
+  rewind() {
+    this.audio.currentTime -= 30;
+  }
+
+  ff() {
+    this.audio.currentTime += 30;
+  }
+
+  seek(e) {
+    this.audio.currentTime = e.target.value;
+  }
+
+  render() {
+    // TODO: Make icons, button text, button labels overrideable?
```

Comment:
There's a few ways you can do this:

- Slots inside the buttons
- Slots outside the buttons (much more complex)
- render props
- overridable methods